cli/command/stack/remove.go

```diff
@@ -77,8 +81,12 @@ func runRemove(dockerCli command.Cli, opts removeOptions) error {
 		}
 
 		hasError := removeServices(ctx, dockerCli, services)
-		hasError = removeSecrets(ctx, dockerCli, secrets) || hasError
-		hasError = removeConfigs(ctx, dockerCli, configs) || hasError
+		if versions.GreaterThanOrEqualTo(version.APIVersion, "1.25") {
+			hasError = removeSecrets(ctx, dockerCli, secrets) || hasError
+		}
+		if versions.GreaterThanOrEqualTo(version.APIVersion, "1.30") {
```

Comment:
These two conditions aren't necessary because `removeSecrets()` and `removeConfigs()` iterate over the array, so it's already a no-op if they are empty.